common/types/activation.go

```diff
@@ -184,6 +186,17 @@ func NewActivationTx(
 	return atx
 }
 
+// Golden returns true if atx is from a checkpoint snapshot.
+// a golden ATX is not verifiable, and is only allowed to be prev atx or positioning atx.
+func (atx *ActivationTx) Golden() bool {
+	return atx.golden
+}
+
+// SetGolden set atx to golden.
+func (atx *ActivationTx) SetGolden() {
+	atx.golden = true
+}
+
```

Comment:
Why not simply make the field public instead? Is it to avoid it being serialized by scale?

Reply:
right. this boolean is inferred.
also, golden atx should never be sync'ed or gossiped